libafl_targets/src/coverage.rs

```diff
@@ -4,5 +4,8 @@ use crate::EDGES_MAP_SIZE;
 
 /// The map for edges.
 pub static mut EDGES_MAP: [u8; EDGES_MAP_SIZE] = [0; EDGES_MAP_SIZE];
+/// The pointer to the map.
+pub static mut EDGES_MAP_PTR: *const u8 = unsafe { EDGES_MAP.as_ptr() };
```

Comment:
Would `pub static mut EDGES_MAP_PTR: *mut u8` work, or does the borrow checker complain?

Reply:
it says it's not stable yet
```
toka@desktop:~/LibAFL/fuzzers/libfuzzer_libpng$ make fuzzer
# Build the libpng libfuzzer library
cargo build
   Compiling libafl_targets v0.5.0 (/home/toka/LibAFL/libafl_targets)
error: `core::slice::<impl [T]>::as_mut_ptr` is not yet stable as a const fn
 --> /home/toka/LibAFL/libafl_targets/src/coverage.rs:8:50
  |
8 | pub static mut EDGES_MAP_PTR: *mut u8 = unsafe { EDGES_MAP.as_mut_ptr() };
  |                                                  ^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

error: could not compile `libafl_targets`
```

Reply:
Hmm this could still be an issue though, since casting `*const` to `*mut` and then accessing is actually undefined behavior. Maybe initialize the MAP_PTR with `null` and later add the edges_map, instead (or something?)
Or maybe we sholud use an UsafeCell, see https://doc.rust-lang.org/std/cell/struct.UnsafeCell.html ?

Reply:
See https://github.com/rust-lang/rust/issues/66136#issuecomment-550003651 for discussion on UB

Reply:
Actually, do we need a ptr here, or could we use a `&mut [u8]`? Could have the same issue WRT `const`, though

Reply:
I feel
> Maybe initialize the MAP_PTR with null and later add the edges_map
is the way to go (possibly in libfuzzer_initialize?)